docs/requirements.txt

```diff
@@ -1,7 +1,3 @@
-matplotlib
```

Comment:
Doesn't make sense to also remove the docs/requirements.txt and put it into the the toml
```yaml
[project.optional-dependencies]
examples =
     ...
docs =
    skmatter[examples]
    sphinx
    sphinx_rtd_theme
    sphinx-gallery
```
and in the .readthedocs.yml we use
```yaml
      extra_requirements:
        - docs
```
(I am not sure if this is the way to make recursive dependencies found it here https://hynek.me/articles/python-recursive-optional-dependencies/)

Reply:
I am always unsure about this. In principle it works putting these dependencies also in the `optional-dependencies`. But, I have the impression that putting docs and test dependencies there is not the correct way. How we test and build our docs is not related to the user.